minifi/minifi-c2/minifi-c2-service/src/main/java/org/apache/nifi/minifi/c2/service/ConfigService.java

```diff
@@ -229,7 +229,7 @@ public Response heartbeat(
                     configuration = configurationProviderValue.getConfiguration();
                 } catch (ConfigurationProviderException cpe) {
                     logger.warn("No flow available for agent class " + agentClass + ", returning No Content (204)");
-                    response = Response.noContent().build();
+                    response = Response.ok(new C2HeartbeatResponse()).build();
```

Comment:
The warning indicates that an HTTP 204 will be returned, but the change returns an empty heartbeat response with an HTTP 200. It seems better in general to return the HTTP 204 if there is no response, but if there is a particular reason for return an empty heartbeat response, the warning log should be adjusted.